src/index.js

```diff
@@ -1,156 +1,83 @@
-/*
- * Copyright (c) 2015 TechnologyAdvice
- */
-'use strict'
-const Promise = require('bluebird')
-const username = require('username')
-const FSEventBridgeClient = require('fs-eventbridge-js')
-const config = require('./lib/config')
-const services = require('./lib/services')
-const proc = require('./lib/process')
-const output = require('./lib/output')
-const parsers = require('./lib/parsers')
-const forwarders = require('./lib/forwarders')
-const url = require('url')
+const _ = require('redash')
+const args = require('./args')
+const config = require('./config')
+const command = require('./command')
+const services = require('./services')
+const proc = require('./proc')
+const output = require('./output')
 
-// run-time manifest (from config)
-let manifest
-// Service links placeholder
-let serviceLinks = []
-// FS-EventBridge client
-let bridge = { stop: () => {} }
-// Remote docker host, if applicable
-let dockerHost
+global.instanceId = require('shortid').generate()
 
-/**
- * Check for and starts services
- * @param {Array} svc Array of services from manifest
- * @returns {Object} promise
- */
-const startServices = svc =>
-  !svc ? Promise.resolve() :
-    services.run(svc)
-      .then(sLinks => {
-        // Create links array for insert into run
-        serviceLinks = sLinks.reduce((links, l) => {
-          return links.concat([ '--link', l ])
-        }, serviceLinks)
+const instance = {
+  /**
+   * @property {number} Timestamp of instance start
+   */
+  startTS: Date.now(),
+  /**
+   * Gets config by merging parsed arguments with config object and returns command
+   * instructions for primaary instance and services
+   * @returns {object} Command instructions
+   */
+  getConfig: () => {
+    const cfg = _.merge(config.load(args.parse().configPath), args.parse())
+    return { services: services.get(cfg), primary: command.get(cfg, 'primary', true) }
+  },
+  /**
+   * Starts services and resolves or rejects
+   * @param {object} cfg Instance config object
+   * @returns {object} promise
+   */
+  startServices: (cfg) => {
+    // No services, resolve
+    if (!cfg.services.length) return Promise.resolve(cfg)
+    // Start services
+    const svcNames = _.map(_.prop('name'), cfg.services).join(', ')
+    const servicesStartSpinner = output.spinner(`Starting service${cfg.services.length > 1 ? 's' : ''} ${svcNames}`)
+    return services.run(cfg.services)
+      .then(() => {
+        servicesStartSpinner.succeed()
+        return cfg
       })
-      .catch(e => {
-        output.error(e)
-        throw 1 // eslint-disable-line no-throw-literal
+      .catch(() => {
+        servicesStartSpinner.fail()
+        throw new Error('Failed to start services')
```

Comment:
No I just meant adapting the error message to singular/plural service(s), the way it's handled in the `Starting` message. (Again, super nitpicky and inconsequential)